tensorboard/webapp/app_routing/effects/app_routing_effects.ts

```diff
@@ -88,7 +104,11 @@ export class AppRoutingEffects {
     this.onInit$,
     this.location.onPopState().pipe(
       map((navigation) => {
-        return {...navigation, browserInitiated: true};
+        return {
+          pathname: this.getAppRootlessPathname(navigation.pathname),
+          replaceState: navigation.replaceState,
+          browserInitiated: true,
+        };
       })
     )
   ).pipe(
```

Comment:
I think things work when the pathname emitted by the 3 source observables on L103 start with a "/", but if navigationRequested is able to emit relative paths like "../experiment/1", my suspicion is that things may break:

- The app's routes are configured without appRoot ("/experiment", "/compare", etc)
- L103, the 3 source observables emit appRootLess pathnames ("/experiment/1") and relative pathnames ("../experiment/1")
- L115 maps those into non-relative pathnames that include the appRoot ("/appRoot/experiment/1")
- L121 uses the routeConfigs to match these strings (that include appRoot) against the app's routes, which is guaranteed to fail to match

Can we add a test case for dispatching something like `navigationRequested({pathname: '../experiments'})` under the presence of path prefix?

Reply:
Great catch. I definitely forgot about the programmatical navigation (which is not currently being used in OSS) and did not properly account for abs/rel path in it. Handled it and added test.